omnibus/files/private-chef-ctl-commands/secrets.rb

```diff
@@ -74,9 +106,41 @@ def confirm_continue!(message)
   end
 end
 
-def set_secret_(group, key, secret)
+def set_secret_(group, key, secret, with_restart=nil)
   credentials.add(group, key, value: secret, frozen: true, force: true)
   credentials.save
+  lookup = "#{group}.#{key}"
+  puts "#{lookup} changed."
+
+  #TODO: do we need to guard against services being empty? (KNOWN_CREDS being out of sync w/ service list)
+  #rather/also-- if a service is not in KNOWN_CREDENTIALS, should it be in the restart list?
+  affected_services = Array(SERVICES_REQUIRING_RESTART[lookup]).select { |s| manage_or_other_service_enabled?(s) }
+
+  return unless affected_services.any?
+
+  service_list = affected_services.sort.join(", ")
+  if with_restart
+    puts "Restarting these services: #{service_list}"
+    affected_services.each { |service| restart_manage_or_other_service(service) }
+  else
+    puts "Please restart these services: #{service_list}"
+  end
+end
+
+def manage_or_other_service_enabled?(service)
+  if service == "chef-manage"
+    File.exist?("/opt/chef-manage/sv/")
+  else
+    service_enabled?(service)
+  end
+end
+
+def restart_manage_or_other_service(service)
+  if service == "chef-manage"
+    Mixlib::ShellOut.new("chef-manage-ctl restart").run_command
+  else
+    run_sv_command_for_service("restart", service)
```

Comment:
We're intentionally avoiding calling `run_sv_service`, since that calls `exit!` and this code potentially will be called more than once. https://github.com/chef/omnibus-ctl/blob/76347dd8e3281928e69e00cca87721d48b0e0e14/lib/omnibus-ctl.rb#L359